src/components/page/page_header/page_header_content.tsx

```diff
@@ -0,0 +1,298 @@
+import { CommonProps } from '../../common';
+import { EuiIcon, EuiIconProps, IconType } from '../../icon';
+import { EuiTab, EuiTabs, EuiTabsProps } from '../../tabs';
+import { Props as EuiTabProps } from '../../tabs/tab';
+import { EuiFlexGroup, EuiFlexItem, EuiFlexGroupProps } from '../../flex';
+import { EuiSpacer } from '../../spacer';
+import { EuiTitle } from '../../title';
+import { EuiText } from '../../text';
+import { useIsWithinBreakpoints } from '../../../services/hooks';
+
+export const ALIGN_ITEMS = ['top', 'bottom', 'center', 'stretch'] as const;
+
+// Gets all the tab props including the button or link props
+type Tab = EuiTabProps & {
+  /**
+   * Visible text of the tab
+   */
+  label: string;
+};
+
+export type EuiPageHeaderContentTitle = {
+  /**
+   * Wrapped in an `H1` so choose appropriately.
+   * A simple string is best
+   */
+  pageTitle?: ReactNode;
+  /**
+   * Optional icon to place to the left of the title
+   */
+  iconType?: IconType;
+  /**
+   * Additional EuiIcon props to apply to the optional icon
+   */
+  iconProps?: Partial<EuiIconProps>;
+};
+
+export type EuiPageHeaderContentTabs = {
+  /**
+   * In-app navigation presented as large borderless tabs.
+   * Accepts an array of `EuiTab` objects;
+   * HELP: This is evaluating to `any[]` in the props table
+   */
+  tabs?: Tab[];
+  /**
+   * Any extras to apply to the outer tabs container.
+   * Extends `EuiTabs`
+   */
+  tabsProps?: Omit<EuiTabsProps, 'size' | 'expand' | 'display'>;
+};
+
+/**
+ * The left side can either be a title with optional description and/or icon;
+ * Or a list of tabs,
+ * Or a custom node
+ */
+type EuiPageHeaderContentLeft = EuiPageHeaderContentTitle &
+  EuiPageHeaderContentTabs & {
+    /**
+     * Position is dependent on existing with a `pageTitle` or `tabs`
+     * Automatically get wrapped in a single paragraph tag inside an EuiText block
+     */
+    description?: string | ReactNode;
+  };
+
+export type EuiPageHeaderContentProps = CommonProps &
+  HTMLAttributes<HTMLDivElement> &
+  EuiPageHeaderContentLeft & {
+    /**
+     * Set to false if you don't want the children to stack at small screen sizes.
+     * Set to `reverse` to display the right side content first for the sack of hierarchy (like global time)
+     */
+    responsive?: boolean | 'reverse';
+    /**
+     * Vertical alignment of the left and right side content;
+     * Default is `middle` for custom content, but `top` for when `pageTitle` or `tabs` are included
+     */
+    alignItems?: typeof ALIGN_ITEMS[number];
+    /**
+     * Pass custom an array of content to this side usually up to 3 buttons.
+     * The first button should be primary, usually with `fill` and will be visually displayed as the last item,
+     * but first in the tab order
+     */
+    rightSideItems?: ReactNode[];
+    /**
+     * Additional EuiFlexGroup props to pass to the container of the `rightSideItems`
+     */
+    rightSideGroupProps?: Partial<EuiFlexGroupProps>;
```

Comment:
```suggestion
    rightSideGroupProps?: Partial<Omit<EuiFlexGroupProps, 'responsive' | 'alignItems'>>;
```

Maybe add `wrap` to the list, also? Unless you want it to be override-able (L249)

Reply:
Yes, I do want them to be overridable. I'm just setting some good defaults, but in the end this is was the easiest path to make the whole group customizable.